### ISSUE TITLE
apigateway get-method --http-method documentation

### ISSUE THREAD

**Issue by rvandegrift** [aws/aws-cli#1620]
What should be specified for --http-method when calling
$ aws apigateway --rest-api-id $apiid --resource-id $resourceid --http-method ... ?

Help says: "Specifies the put method request's HTTP method type."  That's hard to parse!  I guess it's asking me to specify GET, POST, HEAD, etc - but none of those are accepted.

**Comment by kyleknap** [comment]
So with a resource it may have multiple methods depending on the HTTP method type. The ``--http-method`` allows you to grab the method on a resource with that specific HTTP method type (whether it is GET, POST, HEAD). Let me know if that makes sense.

I can see why that is confusing. We are going to make an effort to improve the docs and get some examples because the docs that we do have are at time not clear and pretty minimal.

**Comment by rvandegrift** [comment]
That's what I thought, but it doesn't work for me:
```
$ aws apigateway get-method --rest-api-id $apiid --resource-id $resid --http-method POST

A client error (NotFoundException) occurred when calling the GetMethod operation: Invalid Method identifier specified
$ aws apigateway get-method --rest-api-id $apiid --resource-id $resid --http-method GET

A client error (NotFoundException) occurred when calling the GetMethod operation: Invalid Method identifier specified
$ aws --version
aws-cli/1.9.4 Python/3.4.2 Linux/4.2.0-0.bpo.1-amd64 botocore/1.3.4
```

**Comment by kyleknap** [comment]
You must not have a ``POST`` method for that resource. So try something like this:

First describe your resource:
```
$ aws apigateway get-resource --rest-api-id jqw4a9kno5 --resource-id 7elabk
{
    "resourceMethods": {
        "GET": {}
    }, 
    "pathPart": "foo", 
    "parentId": "z4dkqcfbdl", 
    "path": "/foo", 
    "id": "7elabk"
}
```

Then using the available resource methods listed, try describing one of them:
```
$ aws apigateway get-method --rest-api-id jqw4a9kno5 --resource-id 7elabk --http-method GET
{
    "apiKeyRequired": false, 
    "httpMethod": "GET", 
    "authorizationType": "NONE", 
    "methodIntegration": {
        "cacheKeyParameters": [], 
        "httpMethod": "GET", 
        "type": "HTTP", 
        "uri": "https://foo.endpoint.com", 
        "cacheNamespace": "7elabk"
    }
}
```
If you try to describe a method that does not exist, say ``POST``, you will get an error:
```
$ aws apigateway get-method --rest-api-id jqw4a9kno5 --resource-id 7elabk --http-method POST

A client error (NotFoundException) occurred when calling the GetMethod operation: Invalid Method identifier specified
```

Hope this helps.

**Comment by rvandegrift** [comment]
Excellent - I just misunderstood the error message.  Thanks a ton!

**Comment by dayglojesus** [comment]
Docs are still an issue and when the method is `ANY` the `get-resource` operation returns no `resourceMethods` key and the `get-method` operation produces the error mention above.

**Comment by JekxDevil** [comment]
Issue still not solved using boto3 API `get_method()` in python3.9 with same [error code](https://github.com/aws/aws-cli/issues/1620#issuecomment-1699223641) reported by @shagul998 .
